### PR TITLE
feat: add NASA Earthdata QGIS tool factory

### PR DESCRIPTION
**Pull request by giswqs**
## Summary
- Add `for_nasa_earthdata` factory and `earthdata_tools` adapter exposing catalog search, CMR granule search, footprint display, raster loading, and plugin-dock helpers.
- Extend the QGIS GUI-thread guard in `GeoAgent.chat` so synchronous chat is blocked for the `nasa_earthdata` integration as well as `nasa_opera`.
- Add tests covering import safety, the empty-iface case, the registered tool surface, and the GUI-thread guard.

## Test plan
- [x] `pytest tests/test_nasa_earthdata_tools.py -q`
- [x] `pre-commit run --all-files`

### CODE REVIEW

File: geoagent/tools/nasa_earthdata.py [hunk 1]
```diff
@@ -1,24 +1,554 @@
...
+    for strategy in ("environment", "netrc"):
+        try:
+            auth = earthaccess.login(strategy=strategy)
+            if getattr(auth, "authenticated", bool(auth)):
+                return
+        except Exception:
+            continue
+    raise RuntimeError(
+        "NASA Earthdata authentication failed. Configure credentials in the "
+        "NASA Earthdata plugin settings, environment, or ~/.netrc."
```
[the remark this answers is not in the document]
Comment (giswqs) [reply geoagent/tools/nasa_earthdata.py]: Done in a3728e3 — reworded the auth-failure RuntimeError to only mention supported sources (explicit args, env vars, ~/.netrc).

File: geoagent/tools/nasa_earthdata.py [hunk 2]
```diff
@@ -1,24 +1,554 @@
...
+    import earthaccess
+
+    username = (username or os.environ.get("EARTHDATA_USERNAME", "")).strip()
+    password = (password or os.environ.get("EARTHDATA_PASSWORD", "")).strip()
+    if username and password:
+        os.environ["EARTHDATA_USERNAME"] = username
+        os.environ["EARTHDATA_PASSWORD"] = password
+        auth = earthaccess.login(strategy="environment")
+        if getattr(auth, "authenticated", bool(auth)):
+            return
```
[the remark this answers is not in the document]
Comment (giswqs) [reply geoagent/tools/nasa_earthdata.py]: Done in a3728e3 — explicit credentials are now scoped to the login call and the previous EARTHDATA_USERNAME/PASSWORD env values are restored in a finally block, so passwords no longer persist in the process environment.

File: geoagent/tools/nasa_earthdata.py [hunk 3]
```diff
@@ -1,24 +1,554 @@
...
+        name="search_earthdata_catalog",
+        available_in=("full", "fast"),
+    )
+    def search_earthdata_catalog(
+        query: str = "",
+        max_results: int = 20,
+        catalog_url: Optional[str] = None,
+    ) -> dict[str, Any]:
+        """Search the NASA Earthdata dataset catalog by short name or title."""
+        rows = _load_catalog_rows(catalog_url)
```
[the remark this answers is not in the document]
Comment (giswqs) [reply geoagent/tools/nasa_earthdata.py:324]: Done in a3728e3 — added tests that monkeypatch urlopen with a small TSV payload to assert filtering/max_results behavior, plus a test that non-HTTPS catalog URLs are rejected before any network call.
